packages/frontend/src/features/sqlRunner/components/ContentPanel.tsx

```diff
@@ -143,55 +126,34 @@ export const ContentPanel: FC = () => {
         [activeEditorTab, selectedChartType],
     );
 
-    const { mutateAsync: runSqlQuery, isLoading } = useSqlQueryRun(
-        projectUuid,
-        {
-            onSuccess: (_data) => {
-                // reset error highlighting
-                setHightlightError(undefined);
-            },
-            onError: ({ error }) => {
-                showToastError({
-                    title: 'Could not fetch SQL query results',
-                    subtitle: error.message,
-                });
-
-                if (error?.data) {
-                    // highlight error in editor
-                    const line = error?.data?.lineNumber;
-                    const char = error?.data?.charNumber;
-                    if (line && char) {
-                        setHightlightError({
-                            line: Number(error.data.lineNumber),
-                            char: Number(error.data.charNumber),
-                        });
-                    }
-                }
-            },
-        },
-    );
-
-    // React Query Mutation does not have a way to keep previous results
-    // like the React Query useQuery hook does. So we need to store the results
-    // in the state to keep them around when the query is re-run.
     const queryResults = useAppSelector(selectSqlQueryResults);
 
     const handleRunQuery = useCallback(
         async (sqlToUse: string) => {
             if (!sqlToUse) return;
-            const newQueryResults = await runSqlQuery({
-                sql: sqlToUse,
-                limit: DEFAULT_SQL_LIMIT,
-            });
 
-            if (newQueryResults) {
-                dispatch(setSqlRunnerResults(newQueryResults));
-            }
-            notifications.clean();
+            await dispatch(
+                runSqlQueryThunk({
+                    sql: sqlToUse,
+                    limit: DEFAULT_SQL_LIMIT,
+                    projectUuid,
+                }),
+            );
         },
-        [runSqlQuery, dispatch],
+        [dispatch, projectUuid],
     );
 
+    useEffect(() => {
```

Comment:
think this is fine. and if there's no error (if it changed and there is a successful query, it clears that toaster)

Reply:
Could this happen when `sqlRun` is fulfilled instead of in a useeffect?

Reply:
no it shouldn't, because this is a react hook 
it happens when `runSqlQuery.reject` happens - an error is set in the store